lib/post_destroyer.rb

```diff
@@ -85,9 +85,9 @@ def destroy
   end
 
   def recover
-    if @user.staff? && @post.deleted_at
+    if (@user.staff? || (user_can_review_topic? && post_is_reviewable?)) && @post.deleted_at
```

Comment:
Minor suggestion, we don't really need two methods for our use case here. I would just fold `post_is_reviewable?` into `user_can_review_post?`

```
def user_can_review_post?
  Guardian.new(@user).can_review_topic?(@post.topic) && Reviewable.exists?(target: @post)
end
```